rfcs/transforms.md

```diff
@@ -0,0 +1,135 @@
+## Motivation
+
+Hasura integrates with external HTTP APIs via a variety of features: actions, event triggers, scheduled triggers. Since the client request that these external APIs expect may be different from what Hasura generates, we want to be able to transform these requests before invoking the external API.
+
+## Spec
+
+We will introduce a key called `transforms` in the configuration for actions, event triggers and scheduled triggers.
+
+The schema for `transforms` is described here:
+
+``` 
+data RequestMethod = GET | POST | PUT | PATCH | DELETE
+
+data Transforms 
+  = Transforms
+  { requestMethod ::  Maybe RequestMethod
+ -- ^ Change the request method to one provided here. Nothing means POST.
+  , requestURL    ::  Maybe TemplateText
+ -- ^ Change the request URL to one provided here. Nothing means original URL.
+  , request_body  ::  Maybe TemplateText                  # Raw templating
+ -- ^ Change the request_body to one provided here. Nothing means original payload. Only applicable for POST, PUT, PATCH methods
+ -- The encoding of the content is defined by the 'request_content_type' field defined below.
+  , request_content_type :: AllowedContentTypes
```

Comment:
@paf31 What if we accept a field called `request_content_type` ? I have added examples of this in examples section.